pkg/module/fetch.go

```diff
@@ -12,7 +12,11 @@ func Fetch(ctx context.Context, s storage.Backend, fetcher Fetcher, mod, version
 		return NewErrModuleExcluded(mod)
 	}
 
-	if s.Exists(ctx, mod, version) {
+	moduleExists, err := s.Exists(ctx, mod, version)
+	if err != nil {
+		return err
```

Comment:
same as above, let's include `const op errors.Op = "module.Fetch"` and then return errors.Op(op, err) for this function.